src/index.ts

```diff
@@ -0,0 +1,33 @@
+import { extendEnvironment } from "@nomiclabs/buidler/config";
+import { readArtifactSync } from "@nomiclabs/buidler/plugins";
+import { BuidlerRuntimeEnvironment } from "@nomiclabs/buidler/types";
+import { ContractFactory, ethers, Signer } from "ethers";
+
+import { EthersProviderWrapper } from "./ethers-provider-wrapper";
+
+declare module "@nomiclabs/buidler/types" {
+  interface BuidlerRuntimeEnvironment {
+    ethers: {
+      provider: EthersProviderWrapper;
+      getContract: (name: string) => Promise<ContractFactory>;
+      signers: () => Promise<Signer[]>;
+    };
+  }
+}
+
+extendEnvironment((env: BuidlerRuntimeEnvironment) => {
+  const wrapper = new EthersProviderWrapper(env.provider);
+  env.ethers = {
+    provider: wrapper,
+    getContract: async (name: string): Promise<ContractFactory> => {
+      const artifact = readArtifactSync(env.config.paths.artifacts, name);
```

Comment:
There's an async version of this function. Using sync things in async functions is somewhat unexpected